rllib/core/rl_trainer/rl_trainer.py

```diff
@@ -234,15 +253,35 @@ def update(self, batch: MultiAgentBatch) -> Mapping[str, Any]:
             A dictionary of results.
         """
         if not self.distributed:
-            fwd_out = self._module.forward_train(batch)
-            loss = self.compute_loss(fwd_out=fwd_out, batch=batch)
-            gradients = self.compute_gradients(loss)
-            post_processed_gradients = self.on_after_compute_gradients(gradients)
-            self.apply_gradients(post_processed_gradients)
+            return self._update(batch)
         else:
-            self.do_distributed_update(batch)
+            return self.do_distributed_update(batch)
+
+    def _update(self, batch: MultiAgentBatch) -> Mapping[str, Any]:
+        # TODO (Kourosh): remove the MultiAgentBatch from the type, it should be
+        # NestedDict from the base class.
+        batch = self._convert_batch_type(batch)
+        fwd_out = self._module.forward_train(batch)
+        loss = self.compute_loss(fwd_out=fwd_out, batch=batch)
+        gradients = self.compute_gradients(loss)
+        post_processed_gradients = self.on_after_compute_gradients(gradients)
```

Comment:
Hmm, now that I see the entire sequence:

Ok, suggestions:
- Either: compute_loss, compute_gradients, postprocess_gradients, apply_gradients, compile_results
- OR: compute_loss, compute_gradients, on_gradients_computed, apply_gradients, on_rl_trainer_update_complete
-- with `on_gradients_computed` and `on_rl_trainer_update_complete` being configurable callbacks (so users don't need to subclass anything just to custom tweak these two behaviors here).

Reply:
ok I'll make it simply `postprocess_gradients()` since I am against using callbacks in this context because I think they increase complexity for new comers.

Reply:
Great, thanks!